### PR TITLE
Handle signals with different cdims in dsp.convolve

### LINKED ISSUE #382: Improve broadcasting in dsp.convolve
## General

* pyfar version: 0.5.1
* Python version: 3.10
* Operating System: all
* Did you install pyfar via pip: yes

## Description

`dsp.convolve` requires signals of the same dimension to work. This is often not the case and we could think about adding broadcasting for `signal.cshape` to the function to make it more versatile.

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Closes #382 

### Changes proposed in this pull request:

- add cdim handling to dsp.convolve
- add testing thereof
- add cdim to audio concept documentation because cdim is now mentioned in the docstring of dsp.convolve

### CODE REVIEW

**Review by ahms5 — APPROVED** [review]

**Review by sikersten — APPROVED** [review]
Looks good, thanks!
Would this be necessary for deconvolve, too? (in a separate PR)

**Comment by f-brinkmann** [comment]
I think everything is fine in deconvolve. Since we manually do that by matrix-multiplication the signals broadcast right away.

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -1577,7 +1580,9 @@ def convolve(signal1, signal2, mode='full', method='overlap_add'):
     Returns
     -------
     Signal
-        The convolution result as a Signal object.
+        The convolution result as a Signal object. The
+        :py:mod:`cdim <pyfar._concepts.audio_classes>` matches the bigger cdim
+        of the two input signals.
```
Comment (mberz) [inline pyfar/dsp/dsp.py:1585]:
```suggestion
        The convolution result. The
        :py:mod:`cdim <pyfar._concepts.audio_classes>` matches the bigger cdim
        of the two input signals.
```
The type of the return data is already given in the line above.

Comment (f-brinkmann) [reply pyfar/dsp/dsp.py]: Yes, done

**Pull request merged**